src/core/sorted_map.cc

```diff
@@ -769,5 +769,19 @@ SortedMap* SortedMap::FromListPack(PMR_NS::memory_resource* res, const uint8_t*
   return zs;
 }
 
+bool SortedMap::DefragIfNeeded(float ratio) {
+  auto cb = [this](sds old_obj, sds new_obj) {
+    score_tree->Delete(old_obj);
```

Comment:
Deletion and Insertion of the entry inside btree is not strictly needed since it's the same entry.
Please add a TODO saying to allow replacing the pointers by adding a special 
`ForceUpdate(KeyT)` method to `BPTree`  that assumes the key is the same.

Reply:
YES this is a gem +100